recipe-generator/src/main/resources/templates/java/ingredient.liquid

```diff
@@ -49,7 +49,9 @@ public class {{ingredientName}} extends {{superclass}} {
         setRequired("{{required.name}}", {{required.default | javavalue:required.type}});
     {%- endfor %}
     }{% endif %}
-
+    public {{ingredientName}}({{ingredientName}} other) {
```

Comment:
The argument is there only for overloading purposes? It's not used...

Can this constructor be private or protected since it is only used in the `duplicate` method?

Reply:
Unfortunately `duplicate()` can't use reflection to invoke a protected constructor. The generated ingredients may belong to some other package that the user specifies.

Reply:
And `duplicate()` has to invoke some constructor in order to create a new instance..

Reply:
It can't be a no-arg constructor?

Reply:
No-arg constructors can be generated by the framework, and can have work in them. For example, if you have 3 required parameters but no initializers, it will create a default constructor and set those properties inside. A dumb copy constructor seemed like a good tradeoff..

Reply:
But maybe I should instead have ingredients call their copy constructor instead of `duplicate()`, and have the constructor call duplicate with just copies the `other`'s properties?

Reply:
I was going to suggest something like that. Something so that `new FooIngredient(other)` does the duplication and doesn't require being wrapped by `other.duplicate()`. Mainly just because the `FooIngredient(FooIngredient)` constructor must be public which makes weird things possible.

Reply:
Can all the `duplicate` logic be placed into the root `Ingredient(Ingredient)` constructor?

Reply:
I'll give that a try.